divi/src/main.cpp

```diff
@@ -3429,7 +3439,13 @@ bool ProcessNewBlock(CValidationState& state, CNode* pfrom, CBlock* pblock, CDis
     //    return error("ProcessNewBlock() : duplicate proof-of-stake (%s, %d) for block %s", pblock->GetProofOfStake().first.ToString().c_str(), pblock->GetProofOfStake().second, pblock->GetHash().ToString().c_str());
 
     // NovaCoin: check proof-of-stake block signature
-    if (!pblock->CheckBlockSignature())
+    // At this point, we do not have context enough to know for sure if the
+    // staking-vault fork (block version 5) is active or not.  Thus we
+    // optimistically accept vault signatures as well, which will still weed
+    // out completely invalid blocks quickly.  Later on in ConnectBlock where
+    // this context is available, we will check again with the proper
+    // flag to make sure the fork is phased in properly.
+    if (!pblock->CheckBlockSignature(true))
         return error("ProcessNewBlock() : bad proof-of-stake block signature");
```

Comment:
We probably could, but all the other code around this place just does verification without the context, and thus I think it would not be a good idea to introduce a "hack" here to access the context again.

It is also not really needed - we will invalidate the block later on anyway, so this is just a quick check to prevent simple DoS attack vectors; for this, the check with `true` is good enough.  Also, later on (after the fork), we can simplify the logic anyway and remove the argument to `CheckBlockSignature` again (so this is just a temporary measure).